### PR TITLE
[ADP-2328] add cbor field to TxRelation

### PR DESCRIPTION
**Pull request by paolino**
- expose a prism to convert from database CBOR records to `TxCBOR` datatype
- add `cbor` field to `TxRelation`

### Issue Number

ADP-2328

### CODE REVIEW

**Comment by paolino** [comment]
bors try

File: lib/wallet/src/Cardano/Wallet/DB/Store/CBOR/Store.hs [hunk 1]
```diff
@@ -1,103 +0,0 @@
...
-        cbors <- selectList [] []
-        pure $ first (SomeException . CBOROutOfEra . snd) $ do
-            ps <- mapM (fromTxCBOR . entityVal) cbors
-            pure . TxCBORSet . Map.fromList $ ps
-    , writeS = \txs -> do
-          repsertCBORs txs
-    , updateS = \_ -> \case
-          Append addendum -> repsertCBORs addendum
-          DeleteTx tid -> deleteWhere [CborTxId ==. tid ]
-    }
```
Comment (HeinrichApfelmus) [inline lib/wallet/src/Cardano/Wallet/DB/Store/CBOR/Store.hs:103]: Oh 😲. I was thinking that we may want to keep the `CBOR` `Store` and `Model` around — they are self-contained, well-tested and may eventually become a drop-in replacement for the `Transactions` `Store` and `Model`. But I leave it up to you.

Comment (paolino) [reply lib/wallet/src/Cardano/Wallet/DB/Store/CBOR/Store.hs:103]: It is there in the git history. We will get it back at that point. Better keep master free of dead code IMO.

File: lib/wallet/src/Cardano/Wallet/DB/Store/Transactions/Store.hs [hunk 2]
```diff
@@ -66,6 +67,8 @@ import Database.Persist.Sql
     , (==.)
     )
 
+import Data.Functor
```
Comment (HeinrichApfelmus) [inline lib/wallet/src/Cardano/Wallet/DB/Store/Transactions/Store.hs]: Style: Explicit imports to the other `import` group.

File: lib/wallet/src/Cardano/Wallet/DB/Store/Transactions/Store.hs [hunk 3]
```diff
@@ -98,24 +101,27 @@ write txs = do
...
 putTxSet:: TxSet -> SqlPersistT IO ()
-putTxSet (TxSet tx_map) = forM_ tx_map $ \TxRelation {..} -> do
-    repsertMany' ins
-    repsertMany' collateralIns
-    repsertMany' $ fst <$> outs
-    repsertMany' $ outs >>= snd
-    repsertMany' $ maybeToList $ fst <$> collateralOuts
-    repsertMany' $ maybeToList (collateralOuts) >>= snd
-    repsertMany' withdrawals
+putTxSet (TxSet tx_map) = do -- forM_ tx_map $ \TxRelation {..} -> do
```
Comment (HeinrichApfelmus) [inline lib/wallet/src/Cardano/Wallet/DB/Store/Transactions/Store.hs]:
```suggestion
putTxSet (TxSet tx_map) = do
```

More batching — looks good. 👍

File: lib/wallet/test/unit/Cardano/Wallet/DB/Store/Transactions/StoreSpec.hs [hunk 4]
```diff
@@ -130,11 +130,16 @@ prop_StoreLaws = withStoreProp $ \runQ ->
         (pure mempty)
         (logScale . genDeltas)
 
+addCBOR :: Tx -> Gen Tx
+addCBOR tx = do
+    mcbor <- arbitrary
+    pure $ tx{txCBOR = mcbor}
```
Comment (HeinrichApfelmus) [inline lib/wallet/test/unit/Cardano/Wallet/DB/Store/Transactions/StoreSpec.hs:136]: 👍  I have checked with

```hs
Test.QuickCheck.sample $ fmap txCBOR . addCBOR =<< arbitrary
```

that this generates `Tx` with interesting `CBOR`.

**Review by HeinrichApfelmus — APPROVED** [review]
Looks good to me! 😊 Minor style comment, happy to merge otherwise! 🔱

**Comment by paolino** [comment]
bors retry

**Pull request merged**